build/parser.mjs

```diff
@@ -942,10 +942,19 @@ class Parser {
    * data from "Ducats or Plat".
    * @param {Item} item data to append vault data to
    * @param {VaultData} vaultData to look up data for the {@param item}
+   * @param {module:warframe-items.Category} category of the data
+   * @param {WikiaData} wikiaData from wikia to apply
    */
-  addVaultData(item, vaultData) {
+  addVaultData(item, vaultData, category, wikiaData) {
+    let vaultCategory = category;
+    if (item.type === 'Archwing') vaultCategory = 'archwings';
+    if (category === 'Upgrades') vaultCategory = 'mods';
+
+    if (!['weapons', 'warframes', 'mods', 'upgrades', 'archwings'].includes(vaultCategory.toLowerCase())) return;
     if (!item.name.endsWith('Prime')) return;
+
     const target = vaultData.find((i) => i.Name.toLowerCase() === item.name.toLowerCase());
+    const wikiaItem = wikiaData[vaultCategory.toLowerCase()].filter((i) => i).find((i) => i.name === item.name);
 
     if (!target) {
```

Comment:
@TobiTenno should wikiaItem be added to this test? So it'll fallback to wikia data in case the ogg data doesn't exist?
```suggestion
    if (!target && !wikiaItem) {
```

Reply:
So wikia doesn't have `EstimatedVaultedDate` or `VaultedDate` so if it doesn't exist it would be better to just skip over it since there's no other way to add it from wikia?

Reply:
Well if it's skipped it won't add the vaulted state right?

Reply:
Yup but it also won't error out when target doesn't exist but Wikia does

Reply:
i see you're right we could just add `target?.ReleaseDate` in that case so it won't error out, because i think being vaulted or not is the priority and not the dates.

Reply:
oooo I didn't know I could use that in JS, this is null check right?

Reply:
yep it just ignores the next dot if it's undefined or null it's called Optional Chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
yeah